### PR TITLE
Refactor rating component

### PR DESCRIPTION
**Pull request by mxth**
This PR is subject to change, based on #50 .

### CODE REVIEW

File: src/modules/rating/rating.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+class SmRatingDirective implements ng.IDirective {
+  static instance(): ng.IDirective {
+    return new SmRatingDirective;
+  }
+
+  restrict = 'E';
+  require = ['smRating', 'ngModel'];
+  controller = SmRatingController;
+  link = (scope, element, attrs, ctrls: [SmRatingController, ng.INgModelController]) => {
```
Comment (m0t0r) [inline src/modules/rating/rating.ts]: can you add typings here ?

File: src/modules/rating/rating.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  }
+}
+
+class SmRatingDirective implements ng.IDirective {
+  static instance(): ng.IDirective {
+    return new SmRatingDirective;
+  }
+
+  restrict = 'E';
+  require = ['smRating', 'ngModel'];
```
Comment (m0t0r) [inline src/modules/rating/rating.ts]: I think we do not need to require `smRating`

File: src/modules/rating/rating.spec.ts [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
   'use strict';
 
-  var $rootScope, $compile, element;
+  let $rootScope, $compile, element;
 
-  beforeEach(module('semantic.ui.components.rating'));
+  beforeEach(angular.mock.module(smRatingModule.name));
 
   beforeEach(inject(function(_$rootScope_, _$compile_) {
     $rootScope = _$rootScope_.$new();
```
Comment (m0t0r) [inline src/modules/rating/rating.spec.ts]: Can you change this to use new scope from $rootScope.$new() and use $injector to get $compile, please ?

**Comment by mxth** [comment]
@m0t0r Do you want me to squash everything before merging?

**Comment by m0t0r** [comment]
@mxth yes, please. Also, remove `demo` folder if you can, please

File: src/modules/rating/rating.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+      .on('mouseleave', () => {
+        this.hoverValue = -1;
+        if (angular.isDefined($attrs.onLeave)) {
+          $scope.$evalAsync($attrs.onLeave);
+        }
+        $element.removeClass(stateHoverParent);
+        this.updateStateHover();
+      })
+      .on('keydown', (evt: any) => {
+        if (/(37|38|39|40)/.test(evt.which)) {
```
Comment (m0t0r) [inline src/modules/rating/rating.ts]: This part should be more generic, probably something like [this](https://github.com/angular/angular/blob/master/modules/angular2_material/src/core/key_codes.ts). That's fine for now

**Pull request merged**